SwitchSpeak-iOS/SwitchSpeak-iOS/TouchGrid/TouchSelection.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  TouchSelection.swift
+//  SwitchSpeak-iOS
+//
+//  Created by Jaspal Singh on 3/21/18.
+//  Copyright © 2018 SwitchSpeak. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class TouchSelection {
+
+	var userId:Int!
+	var settings:UserSettings!
+	var touchGrid:TouchGrid?
+	var phrases:[String]
+	var gridSize: Int
+	var index:Int		//	keeps track of which set of phrases we are currently printing on the gird
+	
+	private var viewContainer:UIView!
+	
+	
+	init(userId:Int, viewContainer:UIView, phrases: [String]) {
+		self.userId = userId
+		self.settings = GlobalSettings.userSettings[userId]
+		self.viewContainer = viewContainer
+		self.touchGrid = TouchGrid(userId: userId, viewContainer: viewContainer)
+		self.phrases = phrases
+		let (rows, cols) = parseRowsAndColumns(gridSize: settings.gridSize)
+		self.gridSize = rows * cols
+		
+		self.index = 0
+		self.refillGrid()
+	}
+	
+	/*
+	*	refills the touch grid with the conetent of the next set of phrases stored in
+	*	the string array 'phrases' starting at index 'index'
+	*/
+	func refillGrid() {
+		
+		let lastIndex = min(phrases.count - 1, index + gridSize - 5)
+		//	for now functional grid cells are recognized by the four phrases mentioned below
+		var gridPhrases = Array(phrases[index...lastIndex]) + ["Oops","Next","Home","Done"]
+		if (gridPhrases.count < gridSize) {	//	add dummy elements
+			//	the string '---' represents a dummy cell in the grid
+			//	need to change this depending on how dumy cells are identified
+			gridPhrases += [String](repeating: "---", count: gridSize - gridPhrases.count)
+		}
+		touchGrid!.resetTouchGrid()
+		touchGrid!.fillTouchGrid(phrases: gridPhrases)
+		index = (index + gridSize - 4) % phrases.count
+	}
+	
+	/*
+	*	This function selects the currently highlighted phrase in the grid, and it returns nil is currently
+	*	a set of hrases are highlighted in the grid
+	*	This function also refills the grid with the next set of phrases incase the 'Next' phrase is selected
+	*	a more suitable function name can be considered since its name is the same as a function in the TouchGrid class
+	*/
+	func makeSelection() -> String? {
```

Comment:
this function name matches another function name in TouchGrid case. Couldn't think of any other suitable name....any suggestions?